Add tooltip to show types of file changes in a commit

uttiya10 opened the pull request:
## Description
Currently, the tooltip shown when hovering over the number of changed files just repeats the same information:
![image](https://user-images.githubusercontent.com/56562649/154863400-354491f1-ee65-4bca-8909-c18721352a1c.png)

This PR changes the tooltip to show the types of files changes in the commit:
![image](https://user-images.githubusercontent.com/56562649/154863504-b4ee08d6-76f5-4231-8c70-41e328f9e602.png)


## Release notes

<!--
You can leave this blank if you're not sure.
If you don't believe this PR needs to be mentioned in the release notes, write "Notes: no-notes".
-->

Notes:

uttiya10 committed:
Add tooltip to show types of changes in a commit

haykam821 commented:
I think that this tooltip should include the icons for each status kind instead of just labels. If implemented, these icons would make the information more glanceable than a simple list.

uttiya10 commented:
![image](https://user-images.githubusercontent.com/56562649/155006357-7d284399-54bc-4f8c-b5fd-3e0f2d2604b5.png)
Are you thinking of something like this?

I would like to try making the text colored (green for added, yellow for modified, etc) but can't seem to get css styling to work inside the tooltip😞

tidy-dev reviewed (changes requested):
@uttiya10 I like the direction you are going with the icons and agree coloring would be desired. There is a property of `tooltipClassName` that you can add to TooltippedContent component. We have been adding css rules to the `_tooltips.scss`. See the `length-hint-tooltip` class for an example. 

Also, need to address what happens on an empty commit. I would suggest to not show a tooltip in this case -> send undefined for the `tooltip` property.
![image](https://user-images.githubusercontent.com/75402236/155215030-656cfb5b-9ad1-4dbd-b89f-3a5e3daab23e.png)

tidy-dev commented on app/src/ui/history/commit-summary.tsx:
```diff
@@ -401,6 +393,48 @@ export class CommitSummary extends React.Component<
+
+    const filesLongDescription = (
+      <>
+        {filesAdded ? <li>{filesAdded} added</li> : null}
```
We prefer explicit checks - (same applies to next two lines)

```suggestion
        {filesAdded > 0 ? <li>{filesAdded} added</li> : null}
```

uttiya10 committed:
Update Design of Changed Files Tooltip

- Tooltip will no longer appear if no files have changed
- Tooltip uses icons and are now coloured

uttiya10 commented:
![image](https://user-images.githubusercontent.com/56562649/155255004-c8cfb1c7-f3a6-49ac-8808-b19f907f212a.png)
Updated the design to use icons and coloured text 🎉

tidy-dev commented:
@uttiya10 This is looking great! One small suggestion. I think it would be more consistent with other tooltips and the changes list if only the icons were colored. 

https://user-images.githubusercontent.com/75402236/155343259-9be2671b-f20f-4d49-a475-9d5cb06828d8.mov

.

ghost commented:
Como faço parar de receber esses emails? Não fiz cadastro nisso, e não aguento mais receber isso a casa 5 mim!!!!! Att, Kellen VidalEm 20 de fev. de 2022 20:47, haykam821 ***@***.***> escreveu:
I think that this tooltip should include the icons for each status kind instead of just labels. If implemented, these icons would make the information more glanceable than a simple list.

—Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

uttiya10 committed:
Only Colour the Icons

uttiya10 commented:
On a completely unrelated note, there seems to be a bug where any errors occurred when trying to trigger a re-run of checks do not appear to the user:

https://user-images.githubusercontent.com/56562649/155421881-af4cb625-b396-410d-ae5c-6df9bdbcdaa7.mp4

tidy-dev reviewed (approved):
✨  Thank you for those changes!

tidy-dev commented:
> On a completely unrelated note, there seems to be a bug where any errors occurred when trying to trigger a re-run of checks do not appear to the user

If you haven't already, would you mind opening an issue for that bug? Also if when you hit rerun, please include in the issue whether if it does trigger a re-run for any of the checks. :)

1766701652 reviewed (commented):
w
